ace-indexer/src/main/java/edu/isi/vista/aceIndexer/IndexAceWithElasticSearch.java

```diff
@@ -0,0 +1,262 @@
+package edu.isi.vista.aceIndexer;
+
+import com.google.common.collect.Lists;
+import edu.isi.nlp.parameters.Parameters;
+import org.apache.http.HttpHost;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.util.Date;
+import java.util.List;
+import java.util.OptionalInt;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Indexes Ace with Elastic Search in a way usable by the external search feature of the Inception
+ * annotator.
+ */
+public class IndexAceWithElasticSearch {
+    //TODO: Modify these completely:
+    private static final String USAGE = "IndexAceWithElasticSearch param_file\n" +
+            "\tparam file consists of :-separated key-value pairs\n" +
+            "\tThe required parameters are:\n" +
+            "\tindexName: the name of the index in a running Elastic Search server to add the documents to\n" +
+            "\tcorpusDirectoryPath: the path to a directory where corpus has been extracted. \n" +
+            "\tformat: annotated_ace, or ace" +
+            "\tcompressed: true if the documents are expected to be compressed, false otherwise" +
+            "\tthreshold: a number between 0 and 1 indicating the percentage of failed indexing doc before terminating program\n" +
+            "Additional parameters can be used to point to an Elastic Search server running somewhere besides the " +
+            "standard ports on localhost. For these, please see the source code.";
+
+
+    private static final Logger log = LoggerFactory.getLogger(IndexAceWithElasticSearch.class);
+
+    private static final String DEFAULT_HOST = "localhost";
+
+    private static final int DEFAULT_PORT_PRI = 9200;
+
+    private static final int DEFAULT_PORT_SEC = 9201;
+
+    private static final String PARAM_CORPUS_DIRECTORY_PATH = "corpusDirectoryPath";
+
+    private static final String PARAM_FORMAT = "format";
+
+    // Still Needed? since the ACE corpus comes as a single .tgz containing all the documents
```

Comment:
This parameter could still be useful to someone who e.g. had downloaded the ACE corpus and `gunzip`-ed it